api/plot/processor.py

```diff
@@ -77,8 +78,11 @@ def process(
             return self.prepare_new_multiline_data_message(params, plot_config)
         elif message.type == MsgType.new_image_data:
             params = message.params
-            if not isinstance(params, ImageData):
-                params = ImageData(**params)
+            if not (isinstance(params, ImageData) or isinstance(params, HeatmapData)):
```

Comment:
As `HeatmapData` is a subclass of `ImageData` then the second `isinstance` is superfluous